### PR TITLE
Fixes the KeyError when the colorizer is presented with a novel log level. (In the test case 5)

### LINKED ISSUE #2362: cocotb colorizer does not support custom logging levels
The attached file demonstrates that cocotb does not support custom logging levels. This may be the intention as there is an assertion in the code regarding acceptable logging levels, it is limiting. 

I don't think cocotb should throw an exception on an unexpected level. Instead I think it should have a default behavior. It appears the log colorization system doesn't know what to do with `5`. It likely needs a `try`/`except`.

```
Traceback (most recent call last):
  File "/Users/raysalemi/opt/anaconda3/lib/python3.8/logging/__init__.py", line 1081, in emit
    msg = self.format(record)
  File "/Users/raysalemi/opt/anaconda3/lib/python3.8/logging/__init__.py", line 925, in format
    return fmt.format(record)
  File "/Users/raysalemi/opt/anaconda3/lib/python3.8/site-packages/cocotb/log.py", line 251, in format
    msg = '\n'.join([SimColourLogFormatter.loglevel2colour[record.levelno] % line for line in msg.split('\n')])
  File "/Users/raysalemi/opt/anaconda3/lib/python3.8/site-packages/cocotb/log.py", line 251, in <listcomp>
    msg = '\n'.join([SimColourLogFormatter.loglevel2colour[record.levelno] % line for line in msg.split('\n')])
KeyError: 5
Call stack:
```
[bug_example.zip](https://github.com/cocotb/cocotb/files/5842227/bug_example.zip)

### PR DESCRIPTION
**Pull request by raysalemi**
If a user defines a log level the colorizer used to raise a KeyError
Now it simply doesn't provide a color as it does with DEBUG level.
I accessed the dictionary using get()

Fixes #2362

### CODE REVIEW

**Comment by raysalemi** [comment]
I fixed the PEP issue and pushed a commit. Does the system automatically rerun the tests?

File: tests/test_cases/issue_2362/issue_2362.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+"""
+Test that cocotb doesn't die if a new logging level is added
+"""
+import cocotb
+import logging
+
+
+@cocotb.test()
+async def test_alu(dut):
```
Comment (ktbarrett) [inline tests/test_cases/issue_2362/issue_2362.py:9]: Perhaps a more descriptive test name?
```suggestion
async def test_custom_log_level(dut):
```

**Review by ktbarrett — COMMENTED** [review]
We have decided to not do issue-specific tests anymore. I apologize for not making that obvious ahead of time. Could you move the test into `/tests/test_cases/test_cocotb/test_logging.py`?

**Comment by raysalemi** [comment]
Done!

File: cocotb/log.py [hunk 2]
```diff
@@ -248,8 +248,8 @@ def format(self, record):
         msg = record.getMessage()
 
         # Need to colour each line in case coloring is applied in the message
-        msg = '\n'.join([SimColourLogFormatter.loglevel2colour[record.levelno] % line for line in msg.split('\n')])
-        level = (SimColourLogFormatter.loglevel2colour[record.levelno] %
+        msg = '\n'.join([SimColourLogFormatter.loglevel2colour.get(record.levelno,"%s") % line for line in msg.split('\n')])
```
Comment (cmarqu) [inline cocotb/log.py:251]:
```suggestion
        msg = '\n'.join([SimColourLogFormatter.loglevel2colour.get(record.levelno, "%s") % line for line in msg.split('\n')])
```

File: tests/test_cases/test_cocotb/test_logging.py [hunk 3]
```diff
@@ -75,3 +75,12 @@ async def test_logging_default_config(dut):
...
         cocotb_log.level = log_level_prev
+
+
+@cocotb.test()
+async def test_custom_logging_levels(dut):
+    logging.basicConfig(level=logging.NOTSET)
+    logging.addLevelName(5, "SUPER_DEBUG")
+    logger = logging.getLogger("name")
+    logger.setLevel(5)
+    logger.log(5, "SUPER DEBUG MESSAGE!")
```
Comment (cmarqu) [inline tests/test_cases/test_cocotb/test_logging.py:86]: The style checker wants a newline at the end of the file, but I'm not sure if I can suggest that with the GitHub GUI without it adding an empty line.

**Comment by raysalemi** [comment]
Fixed

File: cocotb/log.py (hunk 2 above)
Comment (ktbarrett) [reply cocotb/log.py:251]: We can get this on another pass (with `black` hopefully).

**Review by ktbarrett — APPROVED** [review]

**Review by cmarqu — APPROVED** [review]

**Comment by cmarqu** [comment]
Thanks for this contribution!

**Comment by raysalemi** [comment]
My pleasure!

**Pull request merged**